libpod/container_internal.go

```diff
@@ -2738,3 +2739,122 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
 
 	return nil
 }
+
+func (c *Container) resetHealthCheckTimers(noHealthCheck bool, changedTimer bool, isStartup bool) error {
+	if !c.ensureState(define.ContainerStateCreated, define.ContainerStateRunning) {
```

Comment:
Maybe Paused as well? Do we stop the timers when we pause a container?

Reply:
I haven't found any code that can stop the timer.

Reply:
That's interesting. We should talk about this at standup on Monday. Not something we should solve in this PR but it does sound like a bug.

Reply:
The timer runs even after the container is paused. Systemctl displays an error for a few seconds: 
```
Nov 15 18:08:33 fedora podman[42]: error: container <id> is not running
```
I don't see how this is a big problem. We can discuss it on standup, but on Monday, I am going to midterm. I can create an Issue so we don't forget about it and then we can discuss it.

Reply:
https://github.com/containers/podman/issues/24590